crates/chain/src/chain_data.rs

```diff
@@ -10,7 +10,7 @@ pub enum ChainPosition<A> {
     /// The chain data is seen as confirmed, and in anchored by `A`.
     Confirmed(A),
     /// The chain data is not confirmed and last seen in the mempool at this timestamp.
-    Unconfirmed(u64),
+    Unconfirmed(Option<u64>),
```

Comment:
Would it be fair to say that unconfirmed transactions without a last-seen will not have a `ChainPosition`? Would it make sense to rename `ChainPosition` to `CanonicalPosition`?